drivers/gpio/gpio_mspm0.c

```diff
@@ -0,0 +1,311 @@
+	}
+
+	return 0;
+}
+
+static int gpio_mspm0_manage_callback(const struct device *port, struct gpio_callback *callback,
+				      bool set)
+{
+	struct gpio_mspm0_data *data = port->data;
+
+	return gpio_manage_callback(&data->callbacks, callback, set);
+}
+
+static uint32_t gpio_mspm0_get_pending_int(const struct device *port)
+{
+	const struct gpio_mspm0_config *config = port->config;
+
+	return DL_GPIO_getPendingInterrupt(config->base);
+}
+
+static void gpio_mspm0_isr(const struct device *port)
+{
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(gpioa), okay)
+	const struct device *dev_a = DEVICE_DT_GET(GPIOA_NODE);
+	struct gpio_mspm0_data *data_a = dev_a->data;
+	const struct gpio_mspm0_config *config_a = dev_a->config;
+
+	uint32_t status_a = DL_GPIO_getRawInterruptStatus(config_a->base, 0xFFFFFFFF);
+
+	DL_GPIO_clearInterruptStatus(config_a->base, status_a);
+
+	gpio_fire_callbacks(&data_a->callbacks, dev_a, status_a);
+#endif /* DT_NODE_HAS_STATUS(DT_NODELABEL(gpioa), okay) */
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(gpiob), okay)
+	const struct device *dev_b = DEVICE_DT_GET(GPIOB_NODE);
+	struct gpio_mspm0_data *data_b = dev_b->data;
+	const struct gpio_mspm0_config *config_b = dev_b->config;
+
+	uint32_t status_b = DL_GPIO_getRawInterruptStatus(config_b->base, 0xFFFFFFFF);
+
+	DL_GPIO_clearInterruptStatus(config_b->base, status_b);
+
+	gpio_fire_callbacks(&data_b->callbacks, dev_b, status_b);
+#endif /* DT_NODE_HAS_STATUS(DT_NODELABEL(gpiob), okay) */
+}
+
+static bool init_irq = true;
+
+static int gpio_mspm0_init(const struct device *port)
+{
+	/* Powering up of GPIOs is part of soc.c */
+
+	if (init_irq) {
+
+		init_irq = false;
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(gpiob), okay)
+		IRQ_CONNECT(DT_IRQN(GPIOB_NODE), DT_IRQ(GPIOB_NODE, priority), gpio_mspm0_isr,
+			    DEVICE_DT_GET(GPIOB_NODE), 0);
+		irq_enable(DT_IRQN(GPIOB_NODE));
+#elif DT_NODE_HAS_STATUS(DT_NODELABEL(gpioa), okay)
+		IRQ_CONNECT(DT_IRQN(GPIOA_NODE), DT_IRQ(GPIOA_NODE, priority), gpio_mspm0_isr,
+			    DEVICE_DT_GET(GPIOA_NODE), 0);
+		irq_enable(DT_IRQN(GPIOA_NODE));
+#endif
+	}
+
+	return 0;
+}
+
+#ifdef CONFIG_GPIO_GET_DIRECTION
+static int gpio_mspm0_port_get_direction(const struct device *port, gpio_port_pins_t map,
+					 gpio_port_pins_t *inputs, gpio_port_pins_t *outputs)
+{
+	return -ENOTSUP;
+}
+#endif /* CONFIG_GPIO_GET_DIRECTION */
+
+static const struct gpio_driver_api gpio_mspm0_driver_api = {
+	.pin_configure = gpio_mspm0_pin_configure,
+	.port_get_raw = gpio_mspm0_port_get_raw,
+	.port_set_masked_raw = gpio_mspm0_port_set_masked_raw,
+	.port_set_bits_raw = gpio_mspm0_port_set_bits_raw,
+	.port_clear_bits_raw = gpio_mspm0_port_clear_bits_raw,
+	.port_toggle_bits = gpio_mspm0_port_toggle_bits,
+	.pin_interrupt_configure = gpio_mspm0_pin_interrupt_configure,
+	.manage_callback = gpio_mspm0_manage_callback,
+	.get_pending_int = gpio_mspm0_get_pending_int,
+#ifdef CONFIG_GPIO_GET_DIRECTION
+	.port_get_direction = gpio_mspm0_port_get_direction,
```

Comment:
Leaving without filled with return `-ENOSYS` from gpio_port_get_direction.

Reply:
I think that's the idea, isn't it?

Reply:
https://github.com/zephyrproject-rtos/zephyr/pull/79673/files#diff-03b58da585a428ec400f1ae882886df2e644bcb25cad1e998fe3128781f7df31R267 returning ENOTSUP with the function pointer is not needed. IMO we can leave it unfilled here

Reply:
Oh, so just remove entirely since not implementing. I understand.